src/test/java/com/google/edith/LoginServiceTest.java

```diff
@@ -0,0 +1,123 @@
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public final class LoginServiceTest {
+
+  private Map<String, Object> map =
+      ImmutableMap.of("com.google.appengine.api.users.UserService.user_id_key", "12345");
+
+  private LocalServiceTestHelper testHelper =
+      new LocalServiceTestHelper(
+              new LocalDatastoreServiceTestConfig(), new LocalUserServiceTestConfig())
+          .setEnvAttributes(map)
+          .setEnvIsLoggedIn(true)
+          .setEnvAuthDomain("gmail")
+          .setEnvIsAdmin(true)
+          .setEnvEmail("user@gmail.com");
+
+  private LoginService loginService;
+  private final UserService userService = UserServiceFactory.getUserService();
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    testHelper.setUp();
+    loginService = new LoginService(userService, datastore);
+  }
+
+  @After
+  public void tearDown() {
+    testHelper.tearDown();
+  }
+
+  @Mock HttpServletRequest request;
+
+  // Checks the login status of the user.
+  @Test
+  public void checkUserLoggedIn_returnsUserLoggedInStatus() {
+    assertTrue(loginService.checkUserLoggedIn());
+    assertTrue(userService.isUserLoggedIn());
+  }
+
+  // Checks if the entity is stored correctly in Datatstore.
+  @Test
+  public void storeUserInfoEntityInDatastore_addsOneUserInfoEntity() {
+    retrieveStubUserInfo(request);
+    loginService.storeUserInfoEntityInDatastore(request);
+    assertEquals(
+        1,
+        datastore.prepare(new Query("UserInfo")).countEntities(FetchOptions.Builder.withLimit(10)));
+  }
+
+  // Checks that only one entity is created for a user.
+  @Test
+  public void storeUserInfoEntityInDatastore_ifSameEntityExists_doNotStore() {
+    retrieveStubUserInfo(request);
+    // Call storeUserInfoEntityInDatastore() twice to
+    // mimic storing UserInfo entity twice for same user.
+    loginService.storeUserInfoEntityInDatastore(request);
+    loginService.storeUserInfoEntityInDatastore(request);
+    assertEquals(
+        1,
+        datastore.prepare(new Query("UserInfo")).countEntities(FetchOptions.Builder.withLimit(10)));
+  }
+
+  // Checks if the JSON created has all the user info fields.
+  @Test
+  public void createJsonOfUserInfo_containsAllFieldsOfUserInfo() {
+    retrieveStubUserInfo(request);
+
+    String userInfo = loginService.createJsonOfUserInfo();
+
+    assertTrue(userInfo.contains("firstName"));
+    assertTrue(userInfo.contains("lastName"));
+    assertTrue(userInfo.contains("favoriteStore"));
+    assertTrue(userInfo.contains("email"));
+    assertTrue(userInfo.contains("logOutUrl"));
+  }
+
+  // Creates a stub user info for testing.
+  private void retrieveStubUserInfo(HttpServletRequest request) {
```

Comment:
nit: I think maybe "mockUserInfo" or "mockUserInfoFromRequest" or something along those lines that indicates user info in the request is being mocked might be a more descriptive name .

Reply:
Thanks for the suggestion. Changed it to ```mockUserInfoFromRequest```